File: every_language_ever/r/day1.r
processFile = function(filepath) {
  con = file(filepath, "r")
  prev = NULL
  increases = 0
  while ( TRUE ) {
    line = readLines(con, n = 1)
    if ( length(line) == 0 ) {
      break
    }
    cur = strtoi(line)
    if(!is.null(prev) && cur > prev)
    {
        increases = increases + 1
    }
    prev = cur
  }
  print(increases)
  close(con)
}

processFile("input.txt")

File: every_language_ever/go/day1.go
package main

import (
	"fmt" 
	"os"
	"log"
	"bufio"
	"strconv"
)

func main() {
	// declare and initialize in one line without declaring type 
	// woohoo go sacrifice readability but #memory
	file, err := os.Open("input.txt")

	if err != nil{ // really fucking nil, fuck you go
		log.Fatal(err)
	}

	// not sure why I need this
	defer file.Close()

	scanner := bufio.NewScanner(file)

	increases := 0
	prev := 0
	have_prev := 0

	for scanner.Scan(){
		cur, err  := strconv.Atoi(scanner.Text())
		if err != nil{
			log.Fatal(err)
		}
		// Woohoo bad code but it works so not fixing it
		if(have_prev != 0 && cur > prev){
			increases++
			prev = cur
		} else{
			have_prev = 1
			prev = cur
		}
		prev = cur
	}
	fmt.Println(increases)

}